ARKit Example/Classes/ARKit/View/RadarView.m

```diff
@@ -28,33 +28,55 @@ - (id)initAtPoint:(CGPoint)middlePoint {
     return self;
 }
 
-- (void) updatePoints:(ARGeoCoordinate *)centerCoord {
-	double centerAzimuth = centerCoord.azimuth + M_PI / 2;
-	if (centerAzimuth < 0.0) {
-		centerAzimuth = 2 * M_PI + centerAzimuth;
-	}
-	
-	UIImage *dot = [UIImage imageNamed:@"radar_dot.png"];
-	
-	for (UIView *sub in dotsView.subviews) {
-		[sub removeFromSuperview];
-	}
-	
-	for (ARGeoCoordinate *coord in points) {
-		double coordAzimuth = coord.azimuth - centerAzimuth;
-		if (coordAzimuth < 0.0) {
-			coordAzimuth = 2 * M_PI + coordAzimuth;
-		}
-		CGPoint pt;
-		
-		pt.x = dotsView.center.x + cos(coordAzimuth) * coord.radialDistance * self.frame.size.width / (2 * farthest);
-		pt.y = dotsView.center.y + sin(coordAzimuth) * coord.radialDistance * self.frame.size.height / (2 * farthest);
-		
-		UIImageView *point = [[UIImageView alloc] initWithImage:dot];
-		point.center = pt;
-		
-		[dotsView addSubview:point];
-	}
+- (void)updatePoints:(ARGeoCoordinate *)centerCoord reloadData:(BOOL)reloadData {
+    
+    double centerAzimuth = centerCoord.azimuth + M_PI / 2;
+    
+    if (centerAzimuth < 0.0f) {
+        centerAzimuth = 2 * M_PI + centerAzimuth;
+    }
+    
+    // Only remove subviews if reloading the data with new points
+    if (reloadData) {
+        for (UIView *sub in dotsView.subviews) {
+            [sub removeFromSuperview];
+        }
+    }
+    
+    // Keep correct count on object index in loop
+    NSInteger count = 0;
+    
+    for (ARGeoCoordinate *coord in points) {
+        
+        double coordAzimuth = coord.azimuth - centerAzimuth;
+        
+        if (coordAzimuth < 0.0f) {
+            coordAzimuth = 2 * M_PI + coordAzimuth;
+        }
+        
+        CGPoint pt;
+        
+        pt.x = dotsView.center.x + cos(coordAzimuth) * coord.radialDistance * self.frame.size.width / (2 * farthest);
+        pt.y = dotsView.center.y + sin(coordAzimuth) * coord.radialDistance * self.frame.size.height / (2 * farthest);
+        
+        UIImageView *point;
+        
+        // Only create new imageviews if reloaddata or initial load
+        if (points.count != dotsView.subviews.count) {
```

Comment:
All dot views will already be created (from line 40), so we just need to `point = [dotsView.subviews objectAtIndex:count];` here.